hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto

```diff
@@ -519,6 +519,7 @@ message BucketInfo {
     optional int64 quotaInBytes = 15 [default = -2];
     optional int64 quotaInNamespace = 16 [default = -2];
     optional uint64 usedNamespace = 17;
+    optional hadoop.hdds.ECReplicationConfig ecReplicationConfig = 18;
```

Comment:
Based on the previous comment, I would suggest to

 1. Store not only the `ecReplicationConfig` but the `RATIS/STANDALONE` parameters. As we have no backward compatibility issue here we can create an embedded structure which includes `ecReplicationConfig/ratisReplicationConfig/standaloneReplicationConfig` we don't really need to do the magic serialization used elsewhere (ecReplicationConfig is de-serialized from the same structure in proto, but ratisReplicationConfig is de-serialized from `replicationType/replicationFactor`)
 2. I would call it `defaultReplicationConfig` to show that it's nothing more just an (optional) default

Reply:
What do you mean by embedded structure?
you mean to create another proto structure containing ecReplicationConfig/ratisReplicationConfig/standaloneReplicationConfig ? Isn't this force to create proto structures for ratisReplicationConfig/standaloneReplicationConfig as well? IIRC, Currently we have proto structures only for EcReplicationConfig. Not sure this embedded structure confuses ( All other structures have ECREplicationCOnfig, type and  factor. But bucket will have some embedded structure).

Reply:
Discussed with Uma, and we think the best way forward is to create a new structure called:

```
defaultReplicationConfig {
  optional hadoop.hdds.Type type = 1;
  optional hadoop.hdds.Factor factor = 2;
  optional hadoop.hdds.ECReplicationConfig ecReplicationConfig = 3;
}
```
While we do not have a backwards compatibility issue here, it keeps it consistent with the other parts of the code was have change, and we can reuse the same logic to create the internal classes from the proto structures.